pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/opcua/OpcUaClientDriver.java

```diff
@@ -0,0 +1,211 @@
+import org.eclipse.milo.opcua.stack.core.types.builtin.NodeId;
+import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;
+import org.eclipse.milo.opcua.stack.core.types.builtin.unsigned.UInteger;
+import org.eclipse.milo.opcua.stack.core.types.enumerated.BrowseDirection;
+import org.eclipse.milo.opcua.stack.core.types.enumerated.BrowseResultMask;
+import org.eclipse.milo.opcua.stack.core.types.enumerated.NodeClass;
+import org.eclipse.milo.opcua.stack.core.types.enumerated.TimestampsToReturn;
+import org.eclipse.milo.opcua.stack.core.types.structured.*;
+import org.eclipse.milo.opcua.stack.core.util.TypeUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.security.KeyPair;
+import java.security.cert.X509Certificate;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static org.eclipse.milo.opcua.stack.core.types.builtin.unsigned.Unsigned.uint;
+import static org.eclipse.milo.opcua.stack.core.util.ConversionUtil.toList;
+
+public class OpcUaClientDriver extends SimpleMemorylessDriver<OpcUaRawData> {
+
+    private static final Logger log = LoggerFactory.getLogger(OpcUaClientDriver.class);
+    private static final Gson jsonParser = new Gson();
+
+    private static String ENDPOINT = "ENDPOINT";
+    private static String NS_INDEX = "NAMESPACE_INDEX";
+    private static String NODE_ID  = "NODE_IDENTIFIER";
+    private static String NODE_FILTER  = "NODE_FILTER_REGEX";
+    private static OpcUaClient opcUaClient;
+    private static Pattern nodeFilter;
+    private static List<NodeId> sensorList ;
+    private static List<ReadValueId> readValueIds;
+    private static NamespaceTable ns;
+
+    public OpcUaClientDriver(DeviceDriverConfig config) {
+        super(config);
+        try {
+            log.info("Trying to establish connection with OPC server");
+            //TODO :connection monitoring and restore
+            opcUaClient = buildClient();
+            opcUaClient.connect().get();
+            log.info("Connection established with OPC server");
+            ns = opcUaClient.getNamespaceTable();
+            log.info("Creating sensor List");
+            nodeFilter = Pattern.compile(getNodeFilter());
+            sensorList = new LinkedList<>();
+            if (opcUaClient.getAddressSpace().getNode(getNodeID()).getNodeClass().getValue() == NodeClass.Variable.getValue())
+            {
+                sensorList.add(getNodeID());
+            }
+            filterNode(opcUaClient,getNodeID());
+            readValueIds = sensorList.stream().map(nodeId -> new ReadValueId(nodeId, AttributeId.Value.uid(), null, null)).collect(Collectors.toUnmodifiableList());
+        } catch (UaException | InterruptedException | ExecutionException e) {
+            log.error("Error in connection to UA Server", e);
```

Comment:
This is catching an error and logging it but otherwise ignoring it. It would be better to not catch the error at all.

Reply:
Constructor now throws the exception to the caller rather than suppressing it with logging,